include/stl2/detail/iterator/counted_iterator.hpp

```diff
@@ -296,18 +296,34 @@ STL2_OPEN_NAMESPACE {
 
 	namespace ext {
 		template<Iterator I>
+		requires Copyable<I>
 		constexpr auto uncounted(const I& i)
 		noexcept(std::is_nothrow_copy_constructible<I>::value) {
 			return i;
 		}
 
+		template<Iterator I>
+		requires !Copyable<I>
+		constexpr auto uncounted(I&& i) {
+			return std::forward<I>(i);
+		}
+
 		template<class I>
 		constexpr auto uncounted(const counted_iterator<I>& i)
 		STL2_NOEXCEPT_RETURN(
 			i.base()
 		)
 
 		template<Iterator I>
+		requires Copyable<I>
+		constexpr auto recounted(const I&, const I& i, iter_difference_t<I> = 0)
+		noexcept(std::is_nothrow_copy_constructible<I>::value)
+		{
+			return i;
+		}
+
+		template<Iterator I>
+		requires !Copyable<I>
```

Comment:
This requirement shouldn't be necessary: the above overload implies that this one happens when `I` doesn't model `Copyable`.